migrate/v0_8/tendermint/v0_32/genesis.go

```diff
@@ -0,0 +1,96 @@
+package v032
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	//"github.com/pkg/errors" // replaced this pkg with "errors" to avoid adding a dependency
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+	"github.com/tendermint/tendermint/types"
+	tmtime "github.com/tendermint/tendermint/types/time"
+)
+
+const (
+	// MaxChainIDLen is a maximum length of the chain ID.
+	MaxChainIDLen = 50
+)
+
+// GenesisDoc defines the initial conditions for a tendermint blockchain, in particular its validator set.
+type GenesisDoc struct {
+	GenesisTime     time.Time                `json:"genesis_time"`
+	ChainID         string                   `json:"chain_id"`
+	ConsensusParams *ConsensusParams         `json:"consensus_params,omitempty"`
+	Validators      []types.GenesisValidator `json:"validators,omitempty"` // v0.33 GenesisValidator is backwards compatible with v0.32
+	AppHash         tmbytes.HexBytes         `json:"app_hash"`             // moved from `common` to `bytes` as they are the same between v0.32 and v0.33
+	AppState        json.RawMessage          `json:"app_state,omitempty"`
+}
+
+// ValidateAndComplete checks that all necessary fields are present
+// and fills in defaults for optional fields left empty
+func (genDoc *GenesisDoc) ValidateAndComplete() error {
+	if genDoc.ChainID == "" {
```

Comment:
use `strings.TrimSpace`

Reply:
This file is copied in from v0.32 tendermint. My plan was to leave it as close as possible to the original to reduce chance of adding any bugs